src/models/lineChart.js

```diff
@@ -184,7 +184,19 @@ nv.models.lineChart = function() {
                     margin.top = legend.height();
                     availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
                 }
-
+                
+                //Do not display the legend if there isn't enough space
+                if (availableHeight1 === 0) {
+                    g.select('.nv-legendWrap')
+                        .datum([])
+                        .call(legend);
+
+                    if (margin.top != legend.height()) {
+                        margin.top = legend.height();
+                        availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
+                    }
+                }
```

Comment:
This seems like a good idea, but I wonder if there is a way to make this more generic so it can be re-used for other charts too...